Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,43 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(
+        Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        TensorDesc outputCountTensor(DML_TENSOR_DATA_TYPE_UINT32, {1,1,1,1}, builder->GetTensorPolicy()); // Same as input
```

Comment:
Dimensions vary from 1 to 8 (per DirectMLSchema.json), not hardcoded to 4. See Resample.